### PR TITLE
ICU-20910 C API Change Report Tool updates

### PR DESCRIPTION
**Pull request by srl295**
[ICU-20910]

- This commit was used to generate ICU4C 65's report
- This commit was inadvertently left out of [ICU-20162]  #807 

Changes:
- fixup copyright messages
- code cleanup

[ICU-20910]: https://unicode-org.atlassian.net/browse/ICU-20910
[ICU-20162]: https://unicode-org.atlassian.net/browse/ICU-20162

### CODE REVIEW

File: tools/release/java/src/main/java/com/ibm/icu/dev/tools/docs/StableAPI.java [hunk 1]
```diff
@@ -48,72 +48,64 @@
...
-       (For example, copy the ALIASES defination from ICU 3.6
-       Doxygen file to ICU 3.4 Doxygen file.)
-    c) gerenate the XML files
-2. Build the tool
-    Download Apache Xerces Java Parser
-    Build this file with the library
-3. Edit the api-report-config.xml file & Change the file according your real configuration
-4. Run the tool to generate the report.
-
+ * A utility to report the status change between two ICU releases
```
Comment (markusicu) [inline tools/release/java/src/main/java/com/ibm/icu/dev/tools/docs/StableAPI.java]: Please undo the reformatting of this block -- makes it unreadable.

File: tools/release/java/src/main/java/com/ibm/icu/dev/tools/docs/StableAPI.java [hunk 2]
```diff
@@ -489,63 +497,60 @@ static Function fromXml(Node n){
...
          * these are noted as deltas.
          */
-        static private String simplifyList[] = {
-            "[ ]*=[ ]*0[ ]*$", "",      // remove pure virtual - TODO: notify about this difference, separately
-            "[ ]*U_NOEXCEPT", "",       // remove U_NOEXCEPT (this was fixed in Doxyfile, but fixing here so it is retroactive)
-            "[ ]*U_OVERRIDE", "",       // remove U_OVERRIDE
-            "^([^\\* ]+)\\*(.*)::(clone|safeClone|cloneAsThawed|freeze|createBufferClone)\\((.*)", "void*$2::$3($4",
-            "\\s+$", "",                 // remove trailing spaces.
-            "^U_NAMESPACE_END ", "" // Bug in processing of uspoof.h
+        static private String simplifyList[] = { "[ ]*=[ ]*0[ ]*$", "", // remove pure virtual - TODO: notify about this
```
Comment (markusicu) [inline tools/release/java/src/main/java/com/ibm/icu/dev/tools/docs/StableAPI.java]: please split this line after the curly brace

File: tools/release/java/src/main/java/com/ibm/icu/dev/tools/docs/StableAPI.java [hunk 3]
```diff
@@ -645,30 +655,31 @@ private void simplifyPrototype() {
...
             }
         }
-//        private Element toXml(Document doc){
-//            Element  ele = doc.createElement("func");
-//            ele.setAttribute("prototype", prototype);
-//            ele.setAttribute("id", id);
-//            ele.setAttribute("status", status);
-//            return ele;
-//        }
+        // private Element toXml(Document doc){
```
Comment (markusicu) [inline tools/release/java/src/main/java/com/ibm/icu/dev/tools/docs/StableAPI.java]: Please either undo the reformatting of this block or delete it if it's obsolete.

File: tools/release/java/src/main/java/com/ibm/icu/dev/tools/docs/StableAPI.java [hunk 4]
```diff
@@ -726,78 +738,79 @@ static JoinedFunction fromTwoFun(Function left, Function right){
...
-    		throw new InternalError("Couldn't make transformer for " + name);
-    	}
-    	return t;
+        if (is == null) {
+            throw new InternalError("No inputstream set for " + name);
+        }
+        System.err.println("Transforming from: " + name);
+        Transformer t;
+        try {
+            // // check the prolog
```
Comment (markusicu) [inline tools/release/java/src/main/java/com/ibm/icu/dev/tools/docs/StableAPI.java]: ditto mis-formatted block

File: tools/release/java/src/main/java/com/ibm/icu/dev/tools/docs/StableAPI.java [hunk 5]
```diff
@@ -915,30 +929,30 @@ private Node setToNode(Set<JoinedFunction> set) throws ParserConfigurationExcept
...
-        Set<Function> common = new TreeSet<Function>(); //Set<Fun>
+        Set<JoinedFunction> joined = new TreeSet<JoinedFunction>(); // Set<JoinedFun>
+        Set<Function> common = new TreeSet<Function>(); // Set<Fun>
         for (Iterator<Function> iter1 = left.iterator(); iter1.hasNext();) {
             Function f1 = iter1.next();
-//            if (f1.prototype.matches(".*Transliterator::.*")){
-//                System.err.println("left: " + f1.prototype);
-//                System.err.println("left: " + f1.status);
-//            }
+            // if (f1.prototype.matches(".*Transliterator::.*")){
```
Comment (markusicu) [inline tools/release/java/src/main/java/com/ibm/icu/dev/tools/docs/StableAPI.java]: ditto mis-formatted block

File: tools/release/java/src/main/java/com/ibm/icu/dev/tools/docs/StableAPI.java (hunk 1 above)
Comment (srl295) [reply tools/release/java/src/main/java/com/ibm/icu/dev/tools/docs/StableAPI.java]: done

**Comment by markusicu** [comment]
This can probably come out of "draft" and get merged, right?

**Pull request merged**